File: C-白昼夢.go
package main
import "fmt"
import "os"
func main(){
	var S string
	fmt.Scan(&S)
	a := "dream"
	b := "dreamer"
	c := "erase"
	d := "eraser"
	l := len(S)
	S += "00000000000000"
	i := 0
	for i != l {
		if S[i:i + 11] == a + d{
			i += 11
		} else if S[i:i + 13] == b + d{
			i += 13
		} else if S[i: i + 10] == a + c {
			i += 10
		} else if S[i: i + 12] == b + c {
			i += 12
		} else if S[i: i + 5] == a {
			i += 5	

		}else if S[i: i + 7] == b {
			i += 7
		
		
		
		
		
		} else if S[i:i + 6] == d {
			i += 6
		} else if S[i:i + 5] == c {
			i += 5
		} else {
			fmt.Println("NO")
			os.Exit(0)
		}
		
		
	}
	fmt.Println("YES")
}

File: C-白昼夢.py






def main():
    S = input()
    vocabulary_list = ['dream', 'dreamer', 'erase', 'eraser']
    i = 0
    ans = 'YES'
    word = ''
    while(i < len(S)):
        if i + 5 > len(S):
            ans = 'NO'
            break

        if i + 5 == len(S):
            word = S[i: i + 5]
            if word != vocabulary_list[0] and word != vocabulary_list[2]:
                ans = 'NO'
            
            break

        

        if i + 7 == len(S):
            word = S[i: i + 7]
            if word != vocabulary_list[1]:
                ans = 'NO'
            
            break
        
        if i + 6 == len(S):
            word = S[i: i + 6]
            if word != vocabulary_list[3]:
                ans = 'NO'
            
            break

            
        word = S[i: i + 7]
        if word == vocabulary_list[1]:
            if S[i + 7] == 'e' or S[i + 7] == 'd':
                i += 7
                continue

        if word[0: 6] == vocabulary_list[3]:
            if S[i + 6] == 'e' or S[i + 6] == 'd':
                i += 6
                continue
        
        if word[0: 5] == vocabulary_list[0] or word[0: 5] == vocabulary_list[2]:
            i += 5
            continue

        ans = 'NO'
        break

    print(ans)

if __name__ == '__main__':
    main()




        
        


